gammapy/estimators/parameter.py

```diff
@@ -308,6 +308,13 @@ def run(self, datasets, parameter):
 
         with datasets.parameters.restore_status():
 
+            if len(datasets.parameters.free_parameters.names) == 1:
+                if self.reoptimize:
+                    log.warning(
+                        f"No free parameters to reoptimize. Setting reoptimize to False"
+                    )
+                    self.reoptimize = False
```

Comment:
I think modifying the state of the `Estimator` and overwriting the use input is not a good idea.  When the `Estimator` is re-used with different datasets, the state might change arbitrarily...

I think just emitting the warning that `reoptimize` is ignored in this case should be fine.

Reply:
And the fix the actual error as low as possible in the hierarchy, probably in `gammapy.modeling.fit`...

Reply:
> I am still unsure how to fix it in modeling.fit, ie, what should a fit do in the absence of any free parameters? 

I think just emitting the warning here is fine. E.g. [this method]( https://github.com/gammapy/gammapy/blob/master/gammapy/modeling/fit.py#L139) is called at the beginning of each method call of `Fit.` Here we could check:
- If there are free parameters at all, if not emit a warning
- Pass the `reoptimize` option to the method and check inside, whether there are other parameters to re-optimize, if not emit a warning and change the value of reoptimize, like:

```python
datasets, parameters, reoptimize = self._parse_datasets(datasets, reoptimize)
```
Or similar...alternatively this could happen on the `Parameters` level (we already have `Parameters.check_limits()`). E.g. like:

```python
parameters.check_reoptimize()
```